server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -217,8 +217,10 @@ private String convertToString(ArrayList<String> comments) {
 
     for (String comment : comments) {
       comment = comment.replace("\"", "");
-      // comment.replace() could potentially give an empty string,
-      // which messes up lastCharacter.
+      // This does two things:
```

Comment:
This still only tells me what (1) substring() does when given a 0 value, and (2) how 'continue' works. I still don't know why one letter comments are being skipped, since you could've just treated one letter comments specially and skipped the substring() call. This comment should say something like "One letter comments aren't concatenated with the larger string because we don't consider them to be sentences" or "One letter comments can't be concatenated with the other comments because they hurt the overall analysis on the larger string". Just throwing ideas out because I'm not sure what the reason is

Reply:
Ah bet. I see what you're saying now.